paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/sink/cdc/TestTable.java

```diff
@@ -125,14 +130,12 @@ public TestTable(
                 }
 
                 List<CdcRecord> records = new ArrayList<>();
-                boolean shouldInsert = true;
                 if (expected.containsKey(key)) {
                     records.add(new CdcRecord(RowKind.DELETE, expected.get(key)));
-                    expected.remove(key);
-                    // 20% chance to only delete without insert
-                    shouldInsert = random.nextInt(5) > 0;
-                }
-                if (shouldInsert) {
```

Comment:
Here we can retain the old test method. If `isAppendTable  == false`, use the old logic; else use the new logic

Reply:
done.